workflow-testing/src/main/java/com/squareup/workflow1/testing/HeadlessIntegrationTest.kt

```diff
@@ -0,0 +1,189 @@
+package com.squareup.workflow1.testing
+
+import app.cash.turbine.ReceiveTurbine
+import app.cash.turbine.test
+import com.squareup.workflow1.RuntimeConfig
+import com.squareup.workflow1.RuntimeConfigOptions
+import com.squareup.workflow1.Workflow
+import com.squareup.workflow1.WorkflowInterceptor
+import com.squareup.workflow1.renderWorkflowIn
+import com.squareup.workflow1.testing.WorkflowTurbine.Companion.WORKFLOW_TEST_DEFAULT_TIMEOUT_MS
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlin.coroutines.CoroutineContext
+import kotlin.time.Duration.Companion.milliseconds
+
+/**
+ * This is a test harness to run integration tests for a Workflow tree. The parameters passed here are
```

Comment:
If we had had this built alreay, do you think we would have bothered with the `launchForTestingFrom*` machinery?

Reply:
No.